### PR TITLE
Vendor gardener@1.6.5

### PR DESCRIPTION
**Pull request by plkokanov**
**What this PR does / why we need it**:
Vendors gardener 1.6.5 so that the os-ubuntu extension can make use of the Restore and Migrate operation implementations from the operatingsystemconfig generic actuator.

**Which issue(s) this PR fixes**:
Part of [#1631](https://github.com/gardener/gardener/issues/1631)

**Special notes for your reviewer**:

**Release note**:

```improvement operator
Vendor gardener@v1.6.5
```

### CODE REVIEW

**Comment by rfranzke** [comment]
Can you also bump the versions in https://github.com/gardener/gardener-extension-os-ubuntu/blob/master/Dockerfile#L2-L9

File: go.mod [hunk 1]
```diff
@@ -1,24 +1,24 @@
...
 
 replace (
-	k8s.io/api => k8s.io/api v0.16.8 // 1.16.8
-	k8s.io/apiextensions-apiserver => k8s.io/apiextensions-apiserver v0.16.8 // 1.16.8
-	k8s.io/apimachinery => k8s.io/apimachinery v0.16.8 // 1.16.8
-	k8s.io/client-go => k8s.io/client-go v0.16.8 // 1.16.8
-	k8s.io/code-generator => k8s.io/code-generator v0.16.8 // 1.16.8
-	k8s.io/component-base => k8s.io/component-base v0.16.8 // 1.16.8
+	k8s.io/api => k8s.io/api v0.17.6 // 1.17.6
+	k8s.io/apiextensions-apiserver => k8s.io/apiextensions-apiserver v0.17.6 // 1.16.8
```
Comment (ialidzhikov) [inline go.mod:18]: You could remove the comments at the end, they were kind of useful when we were replacing with only commit sha and human readable tags were not supported..

```suggestion
	k8s.io/apiextensions-apiserver => k8s.io/apiextensions-apiserver v0.17.6
```

**Comment by vpnachev** [comment]
/needs rebase
to fix the failing CI check

**Comment by plkokanov** [comment]
Checking what's up with the tests and will fix them.

**Comment by plkokanov** [comment]
Ran tests with a successful migration.

**Pull request merged**